nexus/reconfigurator/planning/src/blueprint_builder.rs

```diff
@@ -149,14 +152,14 @@ pub struct BlueprintBuilder<'a> {
     comments: Vec<String>,
 
     // These fields mirror how RSS chooses addresses for zone NICs.
-    nexus_v4_ips: Box<dyn Iterator<Item = Ipv4Addr> + Send>,
-    nexus_v6_ips: Box<dyn Iterator<Item = Ipv6Addr> + Send>,
+    pub(crate) nexus_v4_ips: AvailableIterator<'static, Ipv4Addr>,
```

Comment:
It makes me nervous to expose this (and the other iterators) to the whole crate.  Can we limit that somehow?  It seems like this is only used by the tests at least.

Reply:
Ah yeah this isn't needed right now, fixed. I needed it to figure out which resources were considered to be in use in the planner, but I'll figure out an alternative way to do that as part of #5552.